core/drivers/crypto/caam/include/caam_utils_sgt.h

```diff
@@ -1,22 +1,60 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 /*
- * Copyright 2018-2019 NXP
+ * Copyright 2018-2021 NXP
  *
  * Brief   Scatter-Gather Table management utilities header.
  */
 #ifndef __CAAM_UTILS_SGT_H__
 #define __CAAM_UTILS_SGT_H__
 
-#include <caam_common.h>
+#include <caam_types.h>
 #include <utee_types.h>
 
+/*
+ * Scatter/Gather Table type for input and output data
+ */
+struct caamsgt {
+	uint32_t ptr_ms;  /* W0 - Address pointer (MS 8 LSBs) */
```

Comment:
This PR will make the CAAM to work only on i.MX. This is something I should have mentioned in the PR description.
Layerscape platforms use another SGT format we've implemented and we are currently stress testing.

@jenswi-linaro Do you want to postpone this PR and wait for the LS implementation to be included? Or you are okay to continue reviewing this PR for now?

Reply:
So with this you're basically breaking Layerscape? If that's the case I think we should postpone this PR since we normally avoid regressions in each commit. I say "avoid" since a good enough reason might justify it. The first two commits still makes sense. Perhaps we can take those separately in a new PR.

Reply:
I have to admit upstreaming the DMA without the Layerscape support might not be a good idea. I will postpone the PR and I will wait for the Layerscape DMA to be ready.
I will submit another PR for the first two commits.

Reply:
I'm adding the LS support to the next PR